### ISSUE TITLE
In a mixed C/Rust project, after opening one Rust file, vscode stops analyzing C code

### ISSUE THREAD

**Issue by cjdelisle** [microsoft/vscode-cpptools#6917]
**Type: LanguageService**

**Describe the bug**
- OS and Version: OSX Big Sur / M1
- VS Code Version: 1.53.0-insider - 2ce26643d85bf985d3d3ce6ce0ff5d788
- C/C++ Extension Version: v1.2.0
- Other extensions you installed (and if the issue persists after disabling them): Rust
- Does this issue involve using SSH remote to run the extension on a remote machine?: no
- A clear and concise description of what the bug is, including information about the workspace (i.e. is the workspace a single project or multiple projects, size of the project, etc).

Unsure if this is the right repo to report on, but mixed Rust/C projects are quite common and useful. If I open only C files then C/C++ extension works as expected, but after opening just one Rust file, intellisense stops working on C files until I close and reopen vscode.

**Steps to reproduce**

1. Install C/C++ and also Rust
2. clone https://github.com/cjdelisle/packetcrypt_rs
3. open a C file such as https://github.com/cjdelisle/packetcrypt_rs/blob/master/packetcrypt-sys/packetcrypt/src/CryptoCycle.c - Observe intellisense works correctly
5. open a rust file such as https://github.com/cjdelisle/packetcrypt_rs/blob/master/packetcrypt-util/src/protocol.rs - Observe the rust file is correctly highlighted
7. Switch back to the C file, observe intellisense is nolonger working on the C file

**Expected behavior**

Visual Studio Code ought to work in mixed-language projects

<details>
  <summary><strong>Logs</strong></summary>

```
Insert logs here.
```
</details>

**Screenshots**

**Additional context**

**Comment by michelleangela** [comment]
@cjdelisle 

Could you provide additional information by sending the following 2 logs?

1. Log output from enabling the [language server](https://code.visualstudio.com/docs/cpp/enable-logging-cpp#_enable-logging-for-the-language-server)
2. Log output from running the command `C/C++: Log Diagnostics`

Also, have you opened this project with any previous version of the C++ extension? If so, did it used to work?

**Comment by cjdelisle** [comment]
Well, I don't know what to say, I just went to make you a nice set of logs with a screen recording and this time everything worked perfectly. I turned logging on and off to make sure that was not the culprit and I checked with multiple mixed projects. I'm using auto-update so there is a possibility this was fixed as a result of some other work that somebody was doing. In any case I'll close this now, this is the fastest I've ever seen a bug I reported get fixed, excellent work 😉 💯
